### DISCUSSION TITLE
po-table - Array.push() não atualiza componente.

### DISCUSSION THREAD

**Discussion by dv8martinez** [po-ui/po-angular#1637] [Tire suas dúvidas]
Pessoal boa tarde,
Estou com uma dúvida:
 - Atualizei meu projeto para a versão mais recente tanto do angular(15.1.4) quanto do po-ui(15.1.0). Antes da atualização se tratando principalmente da po-table, ao adicionar um item no `array(push) `da table, o componente adicionava normalmente e atualizava o html automaticamente, idem para o remover item do `array(splice)`, refletia automaticamente, com o `spread `operator funciona.
 Após a atualização, percebi que esse comportamentos não refletem mais.
 Conseguem me explicar ou me ajudar com isso? Terei substituir todos os push() por spread ?

vlww gente!!

**Comment by alinelariguet** [comment]
Boa tarde @dv8martinez, nas versões mais recentes o push realmente não funciona, para evolução do componente e melhor detecção de erros e manutenção, estamos usando a boa prática da imutabilidade, desta maneira é necessário passar uma nova referência ao componente para que o componente se atualize por completo.
Att.
